src/main/java/com/google/swarm/tokenization/common/SanitizeFileNameDoFn.java

```diff
@@ -74,6 +77,9 @@ public void processElement(ProcessContext c) {
       lastModified = Instant.now().getMillis();
     }
     String fileName = sanitizeFileName(file.getMetadata().resourceId().getFilename());
-    c.outputWithTimestamp(KV.of(fileName, file), Instant.ofEpochMilli(lastModified));
+    if (this.inputProviderType == InputLocation.GCS)
+      c.outputWithTimestamp(KV.of(fileName, file), Instant.ofEpochMilli(Instant.now().getMillis()));
+    else
+      c.outputWithTimestamp(KV.of(fileName, file), Instant.ofEpochMilli(lastModified));
```

Comment:
We probably need a comment here stating the reason why GCS needs special handling.

Reply:
Since the files are being read from pub/sub, the last modified time stamp is earlier than the window when the file is read. Hence, for files coming through pub/sub, the time when the notification is being processed would be the timestamp for the windowing.

Reply:
Done